src/Packlist/PacklistForm.js

```diff
@@ -1,43 +1,55 @@
-import React, { useState, useRef } from 'react'
+import React, { useState, useRef, useEffect } from 'react'
 import { useForm } from 'react-hook-form'
 import { v4 as uuid } from 'uuid'
 import styled from 'styled-components/macro'
 import Input from '../common/Input'
 import Label from '../common/Label'
 import Checkbox from '../common/Checkbox'
 import ErrorMessage from '../common/ErrorMessage'
-import PropTypes from 'prop-types'
 import ListContainer from '../common/ListContainer'
 import ListItem from '../common/ListItem'
 import usePacklists from './usePacklists'
 import reloadIcon from '../img/reloadIcon.svg'
 import saveIcon from '../img/saveIcon.svg'
 import addIcon from '../img/addIcon.svg'
+import { useParams, useHistory } from 'react-router-dom'
+import usePacklistForm from './usePacklistForm'
 
-PacklistForm.propTypes = {
-  onPacklistSave: PropTypes.func.isRequired,
-}
-
-export default function PacklistForm({ onPacklistSave }) {
+export default function PacklistForm() {
   const { register, handleSubmit, reset, errors } = useForm()
-  const [items, setItems] = useState([])
+  const { editPacklistID } = useParams()
   const { packlists } = usePacklists()
+  const packlistToEdit = editPacklist()
+  const [items, setItems] = useState([])
   const [itemError, setItemError] = useState(false)
   const itemRef = useRef(null)
+  const { onPacklistSaveEdit, onPacklistSave } = usePacklistForm()
+  const history = useHistory()
+
   const onSubmit = (packlist, event) => {
     event.preventDefault()
-    // for testing...
-    if (event?.target && typeof event?.target.reset === 'function')
-      //
-      event.target.reset()
-    onPacklistSave({ name: packlist.name, packlist: items, id: uuid })
+    const isExistingPacklist = packlists.some(
+      (existingPacklist) => packlist.name === existingPacklist.name
+    )
+    if (isExistingPacklist) {
+      onPacklistSaveEdit({
+        ...packlist,
+        packlist: items,
+      })
+    } else {
+      onPacklistSave({ name: packlist.name, packlist: items, id: uuid() })
+    }
+    history.push('/packlist/' + packlist.name)
   }
   const handlePacklistKeyDown = (event) => {
     if (event.key === 'Enter') {
       event.preventDefault()
-      addItem({ text: event.target.value, id: uuid() })
+      addItem({ item: event.target.value, itemID: uuid() })
     }
   }
+  useEffect(() => {
+    if (packlistToEdit?.packlist) setItems(packlistToEdit?.packlist)
```

Comment:
I think you don't need the "?" here. If that variable is undefined, the if statement will be false.

Reply:
The useeffect should just be working, when the packlistToEdit is avaiable. I tried it without the ? and got an infinity loop :/